app/controllers/solidus_stripe/intents_controller.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'stripe'
+
+class SolidusStripe::IntentsController < Spree::BaseController
+  include Spree::Core::ControllerHelpers::Order
+
+  before_action :load_payment_method
+
+  def payment_confirmation
+    intent = @payment_method.gateway.request { Stripe::PaymentIntent.retrieve(id: params[:payment_intent]) }
+    payment = @payment_method.find_in_progress_payment_for(current_order)
+
+    unless %w[confirm payment].include?(current_order.state.to_s)
```

Comment:
Is there any scenario where that could be false?

Reply:
I guess you can experiment with checking out from a product page and things like that, or mess with an order using multiple tabs. But as you pointed out below it's mostly defensive, in order to assert expectations and prerequisites.